### PR TITLE
[Benchmark] Throttle on timeout

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
Each time we encounter a timeout we want to throttle unconditionally.

Ref: https://github.com/onflow/flow-go/issues/3548

### CODE REVIEW

**Review by pattyshack — APPROVED** [review]
maybe allow downward (but not upward) adjustment on timeout

File: integration/benchmark/cmd/ci/main.go [hunk 1]
```diff
@@ -400,6 +407,10 @@ func computeTPS(
 	currentTPS := float64(currentTxs-lastTxs) / timeDiff
 	unboundedTPS := uint(math.Ceil(currentTPS))
 
+	if timedout {
```
Comment (gomisha) [inline integration/benchmark/cmd/ci/main.go:410]: can you add a comment here explaining what is happening when there's a timeout?

File: integration/benchmark/cmd/ci/main.go [hunk 2]
```diff
@@ -316,13 +316,16 @@ func adjustTPS(
...
 	lastTxs := uint(lg.GetTxExecuted())
+	lastTimedoutTxs := lg.GetTxTimedout()
 	for {
 		select {
 		// NOTE: not using a ticker here since adjusting worker count in SetTPS
 		// can take a while and lead to uneven feedback intervals.
 		case nowTs := <-time.After(interval):
 			currentSentTxs := lg.GetTxSent()
 			currentTxs := uint(lg.GetTxExecuted())
+			currentTimedoutTxs := lg.GetTxTimedout()
```
Comment (gomisha) [inline integration/benchmark/cmd/ci/main.go:327]: can you add descriptions here explaining what the vars represent? For example, what's the difference between `currentTxs` and `currentTimedoutTxs`?

**Review by gomisha — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
> maybe allow downward (but not upward) adjustment on timeout

I've looked into it, but state machine becomes too complicated to read so I went with a simpler approach.

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**